vyper/semantics/types/utils.py

```diff
@@ -183,7 +183,11 @@ def get_type_from_annotation(
             f"No builtin or user-defined type named '{type_name}'. {suggestions_str}", node
         ) from None
 
-    if getattr(type_obj, "_as_array", False) and isinstance(node, vy_ast.Subscript):
+    if (
+        getattr(type_obj, "_as_array", False)
+        and isinstance(node, vy_ast.Subscript)
+        and not isinstance(node.slice.value, vy_ast.Tuple)
```

Comment:
this line is kind of bugging me. is there some other way of differentiating between dynarray and regular array here?

Reply:
maybe a better way is to wrap the if body in a try-except because dynarray would raise a `StructureException`